### PR TITLE
added fig generation code

### PR DESCRIPTION
**Pull request by Rahul-Brito**
added figure generation code. The test figure I was getting looked weird so I need to maybe debug, but submitting PR for now. Code seems good?

### CODE REVIEW

File: src/b2aiprep/process.py [hunk 1]
```diff
@@ -147,9 +150,10 @@ def extract_opensmile(
 
 def to_features(
     filename: Path,
-    subject: str,
-    task: str,
+    subject: str = None,
+    task: str = None,
     outdir: Path = Path(os.getcwd()),
+    save_figures: bool = True,
```
Comment (satra) [inline src/b2aiprep/process.py:156]: should be added as an option to the CLI command.

File: src/b2aiprep/process.py [hunk 2]
```diff
@@ -172,18 +176,45 @@ def to_features(
...
         "opensmile": features_opensmile,
         "sample_rate": audio.sample_rate,
         "checksum": md5sum,
     }
-    outfile = outdir / f"sub-{subject}_task-{task}_md5-{md5sum}_features.pt"
+    
+    #outfile = outdir / f"sub-{subject}_task-{task}_md5-{md5sum}_features.pt"
+    #saving file with just UUID name for now. TO DO: change it back possibly
+    feat_file_name = Path(filename).stem
+    outfile = outdir / f"{feat_file_name}_features.pt"
```
Comment (satra) [inline src/b2aiprep/process.py:196]:
```suggestion
    if subject is not None:
        if task is not None:
            prefix = f"sub-{subject}_task-{task}_md5-{md5sum}"
        else:
            prefix = f"sub-{subject}_md5-{md5sum}"
    else:            
        prefix = Path(filename).stem
    outfile = outdir / f"{prefix}_features.pt"
```

File: src/b2aiprep/process.py [hunk 3]
```diff
@@ -172,18 +176,45 @@ def to_features(
...
-    return features, outfile
+    
+    if save_figures:
+        #save spectogram as figure
+        
+        #general log spectrogram for image
+        features_specgram_log = specgram(audio, log=True)
+        
+        fig, ax = plt.subplots(figsize = (20,10))
+        ax.matshow(features_specgram_log, origin='lower', aspect=8, norm=LogNorm(vmin=0, vmax=1))
```
Comment (satra) [inline src/b2aiprep/process.py:207]: already getting log specgram.

```suggestion
        ax.matshow(features_specgram_log, origin='lower', aspect=8)
```

File: src/b2aiprep/process.py [hunk 4]
```diff
@@ -172,18 +176,45 @@ def to_features(
...
+        ax.axes.xaxis.set_ticks_position('bottom')
```
Comment (satra) [inline src/b2aiprep/process.py:208]: need to add proper and x and y limits.

**Comment by satra** [comment]
putting tags back so we can release when this is merged.

**Comment by satra** [comment]
in addition to the CLI, which needs to take into account a list of files as opposed to file,subject,task, can you check/update the README so that alex's team can use this on their own with the new csv format?

File: src/b2aiprep/process.py (hunk 4 above)
Comment (Rahul-Brito) [reply src/b2aiprep/process.py:212]: ended up just putting in a better hard-coded image aspect that seems to work across speaking, breathing, coughing

File: src/b2aiprep/process.py (hunk 1 above)
Comment (Rahul-Brito) [reply src/b2aiprep/process.py:160]: think I did it; didn't test batch_convert since I wasn't 100% sure how to test the cli command from my dev branch

**Comment by Rahul-Brito** [comment]
@satra i think this should be ready, maybe i could handle the paths in `batch_csv_convert()` as Path instead of str, and I did not actually run the CLI commands to test

**Comment by satra** [comment]
since we don't have tests yet, please do run the CLI commands as that's what they would be using.

**Comment by Rahul-Brito** [comment]
@satra should be ready to merge

**Pull request merged**